src/util.ts

```diff
@@ -950,6 +950,42 @@ export class Util {
             return className;
         }
     }
+
+    /**
+     * Helper for creating `Range` objects. Prefer using this function because vscode-languageserver's `util.createRange()` is significantly slower
+     */
+    public createRange(startLine: number, startCharacter: number, endLine: number, endCharacter: number): Range {
+        return {
+            start: {
+                line: startLine,
+                character: startCharacter
+            },
+            end: {
+                line: endLine,
+                character: endCharacter
+            }
+        };
+    }
+
+    /**
+     * Create a `Range` from two `Position`s
+     */
+    public createRangeFromPositions(startPosition: Position, endPosition: Position): Range {
+        return {
+            start: startPosition,
```

Comment:
Is it safe to use the object without cloning?

Reply:
yeah I think so. At least in our project, we treat all of the Range and Position objects as read-only.

Reply:
Could that be documented, or could the util have a default-false `clone` option, so that if a plugin author uses the util he would be aware the objects aren't cloned. Otherwise it might be generally safer and not terribly more expensive to avoid using the objects as-is.

Reply:
I don't see any valid use cases for messing with the token positions. However, you're right, it wouldn't be measurably different to just clone, so I'll change that.

Reply:
Done.